File: ios/ClawChatKit/Sources/ClawChatKit/Protocol/PairingMessages.swift
import Foundation

// MARK: - Pair Generate

public struct PairGenerate: BaseMessage, Sendable {
    public let type: String = "pair.generate"
    public let id: String
    public let ts: Int64

    public init() {
        self.id = UUID().uuidString
        self.ts = Int64(Date().timeIntervalSince1970 * 1000)
    }
}

// MARK: - Pair Code

public struct PairCode: BaseMessage, Sendable {
    public let type: String
    public let id: String
    public let ts: Int64
    public let code: String
    public let expiresAt: Int64
}

// MARK: - Device Info

public struct DeviceInfo: Codable, Sendable {
    public let deviceId: String
    public let deviceName: String
    public let platform: AppPlatform
    public let pairedAt: Int64?
    public let lastSeen: Int64?
}

// MARK: - Devices List

public struct DevicesList: BaseMessage, Sendable {
    public let type: String = "devices.list"
    public let id: String
    public let ts: Int64

    public init() {
        self.id = UUID().uuidString
        self.ts = Int64(Date().timeIntervalSince1970 * 1000)
    }
}

public struct DevicesListResponse: BaseMessage, Sendable {
    public let type: String
    public let id: String
    public let ts: Int64
    public let devices: [DeviceInfo]
}

// MARK: - Devices Revoke

public struct DevicesRevoke: BaseMessage, Sendable {
    public let type: String = "devices.revoke"
    public let id: String
    public let ts: Int64
    public let deviceId: String

    public init(deviceId: String) {
        self.id = UUID().uuidString
        self.ts = Int64(Date().timeIntervalSince1970 * 1000)
        self.deviceId = deviceId
    }
}

File: clawchat/android/clawchat-kit/src/main/kotlin/com/clawchat/kit/protocol/PairingMessages.kt
package com.clawchat.kit.protocol

import kotlinx.serialization.Serializable
import java.util.UUID

// MARK: - Pair Generate

@Serializable
data class PairGenerate(
    override val type: String = "pair.generate",
    override val id: String = UUID.randomUUID().toString(),
    override val ts: Long = System.currentTimeMillis(),
) : BaseMessage

// MARK: - Pair Code

@Serializable
data class PairCode(
    override val type: String,
    override val id: String,
    override val ts: Long,
    val code: String,
    val expiresAt: Long,
) : BaseMessage

// MARK: - Device Info

@Serializable
data class DeviceInfo(
    val deviceId: String,
    val deviceName: String,
    val platform: String,
    val lastSeen: Long? = null,
)

// MARK: - Devices List

@Serializable
data class DevicesList(
    override val type: String = "devices.list",
    override val id: String = UUID.randomUUID().toString(),
    override val ts: Long = System.currentTimeMillis(),
) : BaseMessage

// MARK: - Devices List Response

@Serializable
data class DevicesListResponse(
    override val type: String,
    override val id: String,
    override val ts: Long,
    val devices: List<DeviceInfo>,
) : BaseMessage

// MARK: - Devices Revoke

@Serializable
data class DevicesRevoke(
    override val type: String = "devices.revoke",
    override val id: String = UUID.randomUUID().toString(),
    override val ts: Long = System.currentTimeMillis(),
    val deviceId: String,
) : BaseMessage
